Thistle/galley-sdk
# (ENG-1675) Refactor ops menu query

## Description
- Refactors `get_ops_menu_query` and `get_formatted_ops_menu_data` that powers all menu/recipe data Okra receives to reduce inefficiencies caused by querying nested `recipeTreeComponents` and `quantityUnitValues`
- Query pulls in the entire flattened list of `recipeTreeComponents` and uses `ancestorComponentIds` to filter and build a DAG recipe mapping for plating and staging data
- Query replaces `quantityUnitValues` with `unit.UnitValues` for unit conversions


## Test Plan
1. Checkout main
2. Open python shell and run:
```python
from galley.formatted_ops_queries import *; from pprint import pprint

pprint(get_formatted_ops_menu_data(['2023-04-10', '2023-04-13', '2023-04-17', '2023-04-20', '2023-04-24', '2023-04-27'], 'Vacaville'))
```
3. Copy and paste output in a [diff checker](https://www.diffchecker.com/)
4. Checkout this branch
5. Repeat Step 2 and Step 3 to compare the diff

- [ ] Verify `get_formatted_ops_menu_data` in v0.46.0 returns the same shape and data values as it does in v0.45.0


## Versioning
0.45.0 → 0.46.0
